### DISCUSSION TITLE
Delay Fulltext-search?!

### DISCUSSION THREAD

**Discussion by ignoreigor** [paperless-ngx/paperless-ngx#288] [Feature Requests] [answered]
First of all. Congratulations for releasing version 1.6.0. after this long time of halt. I noticed the press announcements on heise.de -  good work. Make paperless-ngx great again!

While my paperless archive is growing, I realized that when document-view is shown, fulltext search starts reacting with the first letter, which has almost no filtering effect. Maybe fulltext-search should start searching when at least a phrase of x keys are entered - or explicit pressing enter.

Well think about it. Best wishes!

**Comment by shamoon** [comment]
This makes sense to me, I'd be happy to implement if others agree? The top search starts with 2 characters, do we think thats good? 3? (Part of me wants them to be the same)

**Reply by shamoon** [reply]
Thanks @tim-vogel Ok , before making a change to how a 'system-wide' feature currently works I would just like to get more input / agreement from a larger group, if other folks have an opinion on this. I'll keep an eye on this for a bit

**Comment by tim-vogel** [comment]
I would also consider this adjustment to be useful.

**Comment by gador** [comment]
I also wouldn't mind if the search starts with the third letter

**Comment by lippoliv** [comment]
I think such search boxes should have a delay after keeps. For example 500ms or 300ms. 

Something like:
```this.searchbox.debounceTime(500).distinctUntilChanged().subscribe(...)```

I'm not sure if the amount of characters should matter.

**Reply by shamoon** [reply]
It already does have a debounce, thats not really the same issue.

**Comment by shamoon** [comment, marked as answer]
Implemented in #401, welcome any feedback over there!
